File: test_api.js
const fetch = require('node-fetch');

async function test() {
  console.log('Testing Node Relay API...');
  const start = Date.now();
  try {
    const res = await fetch('http://localhost:3000/api/v1/chat/completions', {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({
        model: 'anthropic/claude-3-haiku',
        messages: [{role: 'user', content: 'Say exactly: "Relay backend is active!"'}],
        stream: false
      })
    });
    
    console.log('Status:', res.status);
    const data = await res.json();
    console.log(JSON.stringify(data, null, 2));
    console.log(`Took ${(Date.now() - start)/1000}s`);

  } catch(err) {
    console.error('Error:', err);
  }
}

test();

File: test_api.py
import requests
import json
import time

print("Testing API...")
start = time.time()
try:
    response = requests.post(
        "http://localhost:3000/api/v1/chat/completions",
        json={
            "model": "anthropic/claude-sonnet-4-6",
            "messages": [{"role": "user", "content": "Write a 5 sentence poem."}],
            "stream": False
        },
        timeout=30
    )
    print("Response Status:", response.status_code)
    try:
        data = response.json()
        print("Response Body:", json.dumps(data, indent=2))
        print("Time took:", round(time.time() - start, 2), "s")
    except Exception as e:
        print("Raw text:", response.text)
except requests.exceptions.RequestException as e:
    print("Request failed:", str(e))
